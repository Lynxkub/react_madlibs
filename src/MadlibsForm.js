import React , {useState} from 'react';



const MadlibsForm = ({addWords}) => {
const INITAL_STATE = {
    noun: '',
    verb : '',
    adjective : '',
    adverb : ''
};

const [formData , setFormData] = useState(INITAL_STATE);

const handleChange= (e) => {
    const {name ,value } = e.target;
    setFormData(formData => ({
        ...formData,
        [name] : value
    }))
}


const handleSubmit = (e) => {
    e.preventDefault();
    addWords({...formData})
    setFormData(INITAL_STATE);
}


return (
    <div>
        <h3>MadLibs!</h3>
        <form onSubmit = {handleSubmit}>
          
            <input 
            id = 'noun'
            data-testid = 'noun'
            type = 'text'
            name = 'noun'
            placeholder='Noun'
            value = {formData.noun}
            onChange = {handleChange}
            />
            <br />
           
            <input 
            id = 'verb'
            data-testid = 'verb'
            type = 'text'
            name = 'verb'
            placeholder='verb'
            value = {formData.verb}
            onChange = {handleChange}
            />
            <br />
           
            <input 
            id = 'adjective'
            data-testid = 'adjective'
            type = 'text'
            name = 'adjective'
            placeholder='adjective'
            value = {formData.adjective}
            onChange = {handleChange}
            />
            <br />
            
            <input 
            id = 'adverb'
            data-testid = 'adverb'
            type = 'text'
            name = 'adverb'
            placeholder='adverb'
            value = {formData.adverb}
            onChange = {handleChange}
            />
            <br />
            <button>Create Story!</button>
          
           
        </form>
    </div>
)


}


export default MadlibsForm;