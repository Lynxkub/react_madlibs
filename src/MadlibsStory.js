import React from 'react';


const MadlibsStory = ({noun , verb , adjective , adverb , deleteStory}) => {

    const handleClick = () => {
        deleteStory();

    }
    return(
        <div>
            <p>{`The ${adjective} ${noun} likes to ${verb} their friend ${adverb}`}</p>
            <button onClick ={handleClick}>Reset Story</button>
        </div>
    )


}


export default MadlibsStory;