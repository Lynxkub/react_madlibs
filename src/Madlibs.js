import React , {useState} from 'react';
import MadlibsForm from './MadlibsForm';
import MadlibsStory from './MadlibsStory';


const Madlibs = () => {
    // const INITAL_STATE = {
    //     name : '',
    //     verb : '',
    //     adjective : '',
    //     adverb : ''
    // }
    const [words , setWords] = useState('')


   const addWords = (newWords) => {
       setWords(words => [...words , {...newWords}])
   }

   const deleteStory = () => {
       setWords('');
   }
    return (
        <div>
            {words.length === 0 ? 
            <MadlibsForm addWords = {addWords}/> :
            <div>
                <MadlibsStory deleteStory = {deleteStory} noun = {words[0].noun} verb = {words[0].verb} adjective = {words[0].adjective} adverb = {words[0].adverb} />
                
            
            </div>
}
        </div>
    )

}


export default Madlibs;