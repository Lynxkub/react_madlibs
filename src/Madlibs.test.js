import React from 'react';
import {render , fireEvent} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect';
import Madlibs from './Madlibs';
import MadlibsForm from './MadlibsForm';
import MadlibsStory from './MadlibsStory';


it('renders without crashing' , function () {
    render(<Madlibs />)
})

it('matches snapshot' , function () {
    const {asFragment} = render(<Madlibs />);
    expect(asFragment()).toMatchSnapshot();
})

it('created a story' , function () {
    const {getByTestId , queryByText} = render(<Madlibs />);
    const noun = getByTestId('noun');
    const verb = getByTestId('verb');
    const adjective = getByTestId('adjective');
    const adverb = getByTestId('adverb');
    const btn = queryByText('Create Story!')

    // check to make sure a created story does not exist yet
    expect(queryByText('The fast worm likes to race their friend quickly')).not.toBeInTheDocument();

    // enter in all the information and create story

    fireEvent.change(noun , {target : {value : 'worm'}});
    fireEvent.change(verb , {target : {value : 'race'}});
    fireEvent.change(adjective, {target : {value : 'fast'}});
    fireEvent.change(adverb , {target : {value : 'quickly'}});
    fireEvent.click(btn);

    // check to see if the form is no longer in the document

    expect(noun).not.toBeInTheDocument();
    
    // check to see if the story is created correctly

    expect(queryByText('The fast worm likes to race their friend quickly')).toBeInTheDocument();


    // select the reset button and click it to reset the form and story
    const resetBtn = queryByText('Reset Story');
    fireEvent.click(resetBtn);
    expect(queryByText('The fast worm likes to race their friend quickly')).not.toBeInTheDocument();
    const newNounBox = getByTestId('noun')
    expect(newNounBox).toBeInTheDocument();


})