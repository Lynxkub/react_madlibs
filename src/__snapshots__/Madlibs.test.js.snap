// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h3>
        MadLibs!
      </h3>
      <form>
        <input
          data-testid="noun"
          id="noun"
          name="noun"
          placeholder="Noun"
          type="text"
          value=""
        />
        <br />
        <input
          data-testid="verb"
          id="verb"
          name="verb"
          placeholder="verb"
          type="text"
          value=""
        />
        <br />
        <input
          data-testid="adjective"
          id="adjective"
          name="adjective"
          placeholder="adjective"
          type="text"
          value=""
        />
        <br />
        <input
          data-testid="adverb"
          id="adverb"
          name="adverb"
          placeholder="adverb"
          type="text"
          value=""
        />
        <br />
        <button>
          Create Story!
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
